lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -72,12 +75,16 @@ export class TokenCache implements ITokenCache {
             // "clientInfo" from options takes precedence over "clientInfo" in response
             if (options.clientInfo) {
                 this.logger.trace("TokenCache - homeAccountId from options");
-                this.loadIdToken(response.id_token, options.clientInfo, authority.hostnameAndPort, authority.tenant, options);
-                this.loadAccessToken(request, response, options.clientInfo, authority.hostnameAndPort, authority.tenant, options);
+                const cacheRecord = this.loadIdToken(response.id_token, options.clientInfo, authority.hostnameAndPort, authority.tenant, options, authority.authorityType);
+                this.loadAccessToken(request, response, cacheRecord, authority.hostnameAndPort, authority.tenant, options);
+                this.loadRefreshToken(request, response, cacheRecord, authority.hostnameAndPort);
+                return cacheRecord;
             } else if (response.client_info) {
                 this.logger.trace("TokenCache - homeAccountId from response");
-                this.loadIdToken(response.id_token, response.client_info, authority.hostnameAndPort, authority.tenant, options);
-                this.loadAccessToken(request, response, response.client_info, authority.hostnameAndPort, authority.tenant, options);
+                const cacheRecord = this.loadIdToken(response.id_token, response.client_info, authority.hostnameAndPort, authority.tenant, options, authority.authorityType);
+                this.loadAccessToken(request, response, cacheRecord, authority.hostnameAndPort, authority.tenant, options);
```

Comment:
Rather than have the `loadIdToken` return a cache record and then mutating the cache record on calls to `loadAccessToken` and `loadRefreshToken`, could you return the individual entities in each API (i.e. IdTokenEntity/AccessTokenEntity/RefreshTokenEntity) and build the cache record once all are returned?

Here's the 2 reasons I think it'd be better:
1. We prefer to avoid methods that mutate their input parameters to avoid unexpected side effects
2.  It seems like initiating the cache record in `loadIdToken` instead of `loadAccessToken` or `loadRefreshToken` is an arbitrary decision that might confuse future maintainers.

Reply:
Yeah I can see that, I was debating doing it that way originally.

Reply:
I followed this path, let me know how it looks.